src/components/JUPAppBar/JUPAppBar.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { memo } from "react";
+import { styled, alpha } from "@mui/material/styles";
+import { AppBar, Box, Toolbar, IconButton, InputBase } from "@mui/material/";
+import { Menu as MenuIcon, Search as SearchIcon } from "@mui/icons-material";
+import JUPSettingsMenu from "components/JUPSettingsMenu";
+import BlockheightChip from "components/SearchBar/components/BlockheightChip";
+import AddressBook from "components/SearchBar/components/AddressBook";
+import { JUPSidebarWidth } from "utils/common/constants";
+import useBreakpoint from "hooks/useBreakpoint";
+
+interface ISearchAppBarProps {
+  isSidebarExpanded: boolean;
+  toggleFn: () => void;
+}
+
+const SearchAppBar: React.FC<ISearchAppBarProps> = ({ isSidebarExpanded, toggleFn }) => {
+  const isMobileSmall = useBreakpoint("<", "sm");
+
+  return (
+    // TODO: Not keen on this magic number, probably need to box these components out more intelligently so flex can handle this
```

Comment:
it's likely that appbar isn't making a height because it has css properties (or it's children do) that take the automatic height calculation away. aka, the children are floated or absolute positioned or some other display modifying property which isn't allowing the browser to compute a proper height because it's taking the elements out of the layout of the document since the elements aren't changing the documents height.

this means that either a static height needs to be added (aka magic number, like your margin bottom) or looking into adding a 'clear fix' to this box, which will force it to computer a height. it's unclear to me if any side effects will happen with either change, so testing would be required. one possible side effect is that the height gets calculated when the mobile hamburger is clicked, expanding the app bar, and that could cause problems. but testing will need to confirm this even happens.

here is the info on clearfix in a basic setup. https://www.w3schools.com/css/css_float_clear.asp

Reply:
unfortunately clear results in the same issue as not providing a static height (the widgets get tucked underneath/behind the appbar)

Going to resolve this thread for now but i made: https://github.com/Up-North-Coding/jovian/issues/53